### PR TITLE
Add task to remove non-existent track_id from Event.

### PR DESCRIPTION
**Pull request by differentreality**
Fixes https://github.com/openSUSE/osem/issues/718 which seems to still exist in events.o.o
Run with:
`bundle exec rake data:nullify_nonexistent_foreign_keys`

### CODE REVIEW

File: lib/tasks/events.rake [hunk 1]
```diff
@@ -0,0 +1,16 @@
+namespace :events do
+  desc 'Create demo data using our factories'
```
Comment (ChrisBr) [inline lib/tasks/events.rake]: The description is wrong!

**Comment by ChrisBr** [comment]
And we will probably need the same for room and difficulty level.

**Comment by ChrisBr** [comment]
Hm, just checked on production: 
tracks: 1
rooms: 0
difficulty levels: 125

What about in changing the view instead?

**Comment by hennevogel** [comment]
Done in #1004 right?

**Comment by ChrisBr** [comment]
@hennevogel yeah! Though, it's still possible that we have corrupt objects in the database and we get this error on some other places ...

#1004 fixes the exception but not the root cause!

File: lib/tasks/events.rake [hunk 2]
```diff
@@ -0,0 +1,45 @@
+namespace :events do
+  desc 'Nullifies wrong foreign keys for Track, Room, DifficultyLevel'
+
+  task fix_wrong_track: :environment do
```
Comment (ChrisBr) [inline lib/tasks/events.rake]: Why not doing this in one task?

Comment (differentreality) [inline lib/tasks/events.rake]: My logic was 3 different things, 3 different tasks. I have put everything in one task.

File: lib/tasks/events.rake [hunk 3]
```diff
@@ -0,0 +1,44 @@
+namespace :events do
```
Comment (hennevogel) [inline lib/tasks/events.rake]: Please put this into the namespace `:data`

Comment (differentreality) [inline lib/tasks/events.rake]: should that change the filename too?

File: lib/tasks/events.rake [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  desc 'Nullifies wrong foreign keys for Track, Room, DifficultyLevel'
```
Comment (hennevogel) [inline lib/tasks/events.rake]: > desc 'Nullifies wrong foreign keys for Track, Room, DifficultyLevel'
desc 'Nullifies wrong foreign keys'

File: lib/tasks/events.rake [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+    if events.any?
+      puts "Event IDs: #{events.map(&:id)}"
+      events.each do |event|
+        event.track_id = nil
+        event.save!
+      end
+      puts 'Fixed track_id!'
+    end
+
```
Comment (hennevogel) [inline lib/tasks/events.rake]: This really looks like you can DRY it up.

File: lib/tasks/events.rake (hunk 3 above)
Comment (hennevogel) [inline lib/tasks/events.rake]: Sounds like a good idea yes

File: lib/tasks/data.rake [hunk 6]
```diff
@@ -0,0 +1,29 @@
+namespace :data do
+  desc 'Nullify wrong foreign keys'
+
+  task events: :environment do
```
Comment (hennevogel) [inline lib/tasks/data.rake]: Please use a more descriptive task name.

Comment (differentreality) [inline lib/tasks/data.rake]: :nullify_nonexistent_foreign_keys_for_events too long?

Comment (hennevogel) [inline lib/tasks/data.rake]: :nullify_nonexistent_foreign_keys sounds good. Then we can expand it to handle other models in the future :-)

**Comment by hennevogel** [comment]
Cool

**Pull request merged**